test/e2e/edit-app-settings.spec.js

```diff
@@ -0,0 +1,69 @@
+const fs = require('fs');
+const path = require('path');
+const { expect } = require('chai');
+const request = require('request-promise-native');
+
+const { getProjectUrl } = require('./cht-docker-utils');
+const {
+  DEFAULT_PROJECT_NAME,
+  cleanupProject,
+  getProjectDirectory,
+  initProject,
+  runChtConf,
+} = require('./cht-conf-utils');
+
+describe('edit-app-settings', () => {
+  const projectName = DEFAULT_PROJECT_NAME;
+  const projectDirectory = getProjectDirectory(projectName);
+
+  before(async () => {
+    await initProject(projectName);
+  });
+
+  after(async () => {
+    await cleanupProject(projectName);
+  });
+
+  it('disables a language, recompile, and push app settings', async () => {
+    const url = await getProjectUrl(projectName);
+    const initialSettings = await request.get({ url: `${url}/api/v1/settings`, json: true });
```

Comment:
Minor, but seems worth adding a few assertions about the initial state of the settings (`local`, etc) before we edit things.  (Since we are so far removed from how/where these initial settings are being defined).